openy_node/modules/openy_node_blog/src/BlogSimplePaginator.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Drupal\openy_node_blog;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Link;
+use Drupal\Core\Routing\RouteMatchInterface;
+
+class BlogSimplePaginator {
+
+  /**
+   * The node storage.
+   *
+   * @var \Drupal\Core\Entity\EntityStorageInterface
+   */
+  protected $nodeStorage;
+
+  /**
+   * The route match.
+   *
+   * @var \Drupal\Core\Routing\RouteMatchInterface
+   */
+  protected $routeMatch;
+
+  /**
+   * Constructs the BlogSimplePaginator.
+   *
+   * @param EntityTypeManagerInterface $entity_type_manager
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, RouteMatchInterface $current_route_match) {
+    $this->nodeStorage = $entity_type_manager->getStorage('node');
+    $this->routeMatch = $current_route_match;
+  }
+
+  /**
+   * @param $direction
+   *
+   * @return array|mixed[]|null
+   * @throws \Drupal\Core\Entity\EntityMalformedException
+   */
+  private function buildQuery($direction = 'prev') {
+    $currentNode = $this->routeMatch->getParameter('node');
+    $query = $this->nodeStorage->getQuery();
+
+    $compare = $direction == 'prev' ? '<' : '>';
```

Comment:
let's avoid single-line variable definitions. It's very hard to find a variable if such
Let's stick to Drupal styling, it rocks